addon/lib/StudyTelemetryCollector.jsm

```diff
@@ -21,15 +22,266 @@ const BASERESOURCE = "esper-pioneer-shield-study";
+      }
+      if (typeof attributeValue !== "string") {
+        attributeValue = String(attributeValue);
+      }
+      shieldPingPayload[attribute] = attributeValue;
+    }
+
+    return shieldPingPayload;
+
+  }
+
+  /**
+   * These attributes are already sent as part of the telemetry ping envelope
+   * @returns {{}}
+   */
+  static collectTelemetryPingEnvelopeBasedAttributes() {
+
+    const telemetrySessionCurrentPingData = TelemetryController.getCurrentPingData();
+    console.log("telemetrySessionCurrentPingData", telemetrySessionCurrentPingData);
+
+    const environment = telemetrySessionCurrentPingData.environment;
+    const application = telemetrySessionCurrentPingData.application;
+
+    return {
+      "default_search_engine": environment.settings.defaultSearchEngine,
+      "locale": environment.settings.locale,
+      "os": environment.system.os.name,
+      "normalized_channel": application.channel,
+      "profile_creation_date": environment.profile.creationDate,
+      "app_version": application.version,
+      "system.memory_mb": environment.system.memoryMB,
+      "system_cpu.cores": environment.system.cpu.cores,
+      "system_cpu.speed_mhz": environment.system.cpu.speedMHz,
+      "os_version": environment.system.os.version,
+      "system_gfx.monitors[1].screen_width": environment.system.gfx.monitors[0] ? environment.system.gfx.monitors[0].screenWidth : undefined,
+      "system_gfx.monitors[1].screen_width_zero_indexed": environment.system.gfx.monitors[1] ? environment.system.gfx.monitors[1].screenWidth : undefined,
+    };
+
+  }
+
+  /**
+   * Scalars are only submitted if data was added to them, and are only reported with subsession pings.
+   * Thus, we use the current telemetry subsession payload for these attributes
+   * @returns {{}}
+   */
+  static collectTelemetrySubsessionPayloadBasedAttributes() {
+
+    const telemetrySubsessionCurrentPingData = TelemetryController.getCurrentPingData(true);
+    console.log("telemetrySubsessionCurrentPingData", telemetrySubsessionCurrentPingData);
+
+    const payload = telemetrySubsessionCurrentPingData.payload;
+
+    const attributes = {};
+
+    attributes.uptime = payload.simpleMeasurements.uptime;
+    attributes.total_time = payload.simpleMeasurements.totalTime;
+    attributes.profile_subsession_counter = payload.info.profileSubsessionCounter;
+    attributes.subsession_start_date = payload.info.subsessionStartDate;
+    attributes.timezone_offset = payload.info.timezoneOffset;
+
+    // firefox/toolkit/components/places/PlacesDBUtils.jsm
+    // Collect the histogram payloads if present
+    attributes.places_bookmarks_count_histogram = payload.processes.content.histograms.PLACES_BOOKMARKS_COUNT;
+    attributes.places_pages_count_histogram = payload.processes.content.histograms.PLACES_PAGES_COUNT;
+
+    // firefox/browser/modules/BrowserUsageTelemetry.jsm
+    attributes.search_counts = payload.processes.content.histograms.SEARCH_COUNTS;
```

Comment:
Are you sure this histogram is recorded exclusively by the content process? On my machine, for example, I have it only for the "parent" process.

Reply:
Not sure, but Histograms.json notes that it is recording in main and content, but upon inspection, there is not even a processes.parent.histograms key payload: https://www.dropbox.com/s/fauq6gbql0bqr2s/Screenshot%202017-11-02%2018.50.17.png?dl=0

It is not observed neither in processes.content.histograms (https://www.dropbox.com/s/nt96utjtzjta19x/Screenshot%202017-11-02%2018.51.00.png?dl=0) so the code above is the current best guess as to where the information should be. Maybe payload.histograms.SEARCH_COUNTS instead? Advice appreciated on how to address this :)

Reply:
Yes, payload.histograms.SEARCH_COUNTS is the correct way, if you want to record parent process histograms (see [here](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/data/main-ping.html))!

Reply:
Ah, thanks for clarifying, since the linked docs says the opposite :) "payload -> histograms - This section contains histograms and keyed histograms accumulated on content processes." and - where would be the correct place to file this documentation issue?
The clarification necessary would be `histograms: {...},` -> `histograms: {...}, // parent process histograms (content process histograms are found under processes.content.histograms)`